preface

whuan20011 opened the pull request:
Algorithm: dfs
pow(a, b)
learn to change a maximum depth of a recursion

whuan20011 committed:
preface
Algorithm: dfs
pow(a, b)
learn to change a maximum depth of a recursion

smy20011 commented on preface/preface.py:
```diff
@@ -0,0 +1,66 @@
+    res = {}
+    dic = {1:'I', 5:'V', 10:'X', 50:'L', 100:'C', 500:'D', 1000:'M'}
+    roman_numerals(N, 1, dic, res)
+    if 'I' in res:
```
Can be replaced by a for loop

```python
for char in "IVXLCDM":
    if char in res:
        print >>fout, res[char]
```

smy20011 commented on preface/preface.py:
```diff
@@ -0,0 +1,66 @@
+            temp.append(dic[5 * pow(10, t)] + dic[pow(10, t)] + dic[pow(10, t)] + dic[pow(10, t)])
+        if m == 9:
+            temp.append(dic[pow(10, t)] + dic[pow(10, (t + 1))])
+    for p in temp:
```
Can be replaced using python's counter class (https://docs.python.org/2/library/collections.html#collections.Counter)

```python
return collections.Counter(temp)
```

whuan20011 committed:
preface
Algorithm: dfs
change: the print out style

smy20011 commented on preface/preface.py:
```diff
@@ -0,0 +1,66 @@
+def roman_numerals(N, num, dic, res):
+    if num > N:
+        return
+    temp = []
```
Use a string

smy20011 commented on preface/preface.py:
```diff
@@ -0,0 +1,66 @@
+                res[s] = 1
+            else:
+                res[s] += 1
+    roman_numerals(N, a + 1, dic, res)
```
Use for loop instead of recursion.

whuan20011 committed:
preface
Algorithm: changed dfs to while loop
use string instead of list
Important: count the characters in a string by using counter
import collections
res = collections.Counter(string)
